app/screens/selector_screen/selector_screen.tsx

```diff
@@ -31,50 +30,79 @@ import {
+import {UserProfile} from '@mm-redux/types/users';
+import SelectedOptions from './selected_options';
+import {EventSubscription, Navigation} from 'react-native-navigation';
+
+type DataType = DialogOption[] | Channel[] | UserProfile[];
+type Selection = DialogOption | Channel | UserProfile | DialogOption[] | Channel[] | UserProfile[];
+type Props = {
+    actions: {
+        getProfiles: (page?: number, perPage?: number, options?: any) => Promise<ActionResult>,
+        getChannels: (teamId: string, page?: number, perPage?: number) => Promise<ActionResult>,
+        searchProfiles: (term: string, options?: any) => Promise<ActionResult>,
+        searchChannels: (teamId: string, term: string, archived?: boolean | undefined) => Promise<ActionResult>,
+    },
+    getDynamicOptions?: (term: string) => Promise<ActionResult>,
+    currentTeamId: string,
+    data?: DataType,
+    dataSource: string,
+    onSelect: (opt: Selection) => void,
+    isMultiselect?: boolean,
+    selected?: DialogOption[],
+    theme: Theme,
+}
 
-export default class SelectorScreen extends PureComponent {
-    static propTypes = {
-        actions: PropTypes.shape({
-            getProfiles: PropTypes.func.isRequired,
-            getChannels: PropTypes.func.isRequired,
-            searchProfiles: PropTypes.func.isRequired,
-            searchChannels: PropTypes.func.isRequired,
-        }),
-        getDynamicOptions: PropTypes.func,
-        currentTeamId: PropTypes.string.isRequired,
-        data: PropTypes.arrayOf(PropTypes.object),
-        dataSource: PropTypes.string,
-        onSelect: PropTypes.func.isRequired,
-        theme: PropTypes.object.isRequired,
-    };
+type State = {
+    data: DataType | {id: string; data: DataType}[],
+    loading: boolean;
+    searchResults: DialogOption[],
+    term: string;
+    multiselectSelected: DataType,
+}
 
+export default class SelectorScreen extends PureComponent<Props, State> {
     static contextTypes = {
         intl: intlShape.isRequired,
     };
 
-    constructor(props) {
+    private navigationEventListener?: EventSubscription;
+
+    private searchTimeoutId = 0;
+    private page = -1;
+    private next: boolean;
+    private searchBarRef = React.createRef<SearchBar>();
+    constructor(props: Props) {
         super(props);
 
-        this.searchTimeoutId = 0;
-        this.page = -1;
         this.next = props.dataSource === ViewTypes.DATA_SOURCE_USERS || props.dataSource === ViewTypes.DATA_SOURCE_CHANNELS || props.dataSource === ViewTypes.DATA_SOURCE_DYNAMIC;
 
-        let data = [];
+        let data: DataType = [];
         if (!props.dataSource) {
-            data = props.data;
+            data = props.data || [];
+        }
+
+        let multiselectSelected: DataType = [];
+        if (props.isMultiselect && props.selected && !([ViewTypes.DATA_SOURCE_USERS, ViewTypes.DATA_SOURCE_CHANNELS].includes(props.dataSource))) {
+            multiselectSelected = props.selected;
```

Comment:
Just to be sure, this means you cannot initialize a multiselect list of users or channels right?

Reply:
Exactly. We should create a follow-up ticket to support it, but before we need to support this on webapp (user and channel multiselect is not yet supported).